### PR TITLE
Commerce 12528

### PR DESCRIPTION
**Pull request by brianikim**
https://liferay.atlassian.net/browse/COMMERCE-12528

@andrea-ale-sbarra ,
tied to LPP.

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/util/index.js [hunk 1]
```diff
@@ -224,7 +224,7 @@ export function parseOptions(jsonString) {
 	}
 
 	return Array.isArray(options)
-		? options.map(({value}) => `${value}`).join(', ')
+		? options.map(({name}) => `${name}`).join(', ')
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/util/index.js:227]: This doesn't look right, what if I have a text option? I see only the text option name without the value that I set?

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
This doesn't look to work see attachment:
Screenshot 2023-10-03 at 16 07 46
I still see the product option value.
Also my understanding was that you would have added the option name to the cart item json property, is that still true?

Thank you!

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/util/index.js (hunk 1 above)
Comment (brianikim) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/util/index.js:227]: I cannot alter the value implementation because it is too embedded in our code.

Comment (brianikim) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/util/index.js:227]: If you try your reproduction steps again, but change the name of the product option values to something more visible like "6name", etc. You will see the correct results of the PR.

**Comment by brianikim** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/util/index.js (hunk 1 above)
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/util/index.js:227]: Screenshot 2023-10-04 at 11 53 49

I got the point of showing the optionvalue name, and that's fine, but as I thought, this fix is not working for all of the option type, in the screenshot it's failing on text type. Can you please verify what happens with all of the option type and also please sync your solution with COMMERCE-9599, because it changes quite a bit the options in the cart item component.

Thank you!

Comment (brianikim) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/util/index.js:227]: @andrea-ale-sbarra ,

I should have a solution soon. You are right with your testing. There are lot of edge cases which I should be able to take care of.

Sincerely,
Brian I. Kim

**Comment by brianikim** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/util/index.js (hunk 1 above)
Comment (brianikim) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/util/index.js]: See updated commits @andrea-ale-sbarra , tested with COMMERCE-9599 included.

Comment (brianikim) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/util/index.js]: See https://liferay.atlassian.net/browse/COMMERCE-12528 for limitation on Boolean.
